tests/perf-system/submitter/submit.cpp

```diff
@@ -41,43 +41,95 @@ void read_parquet_file(string generator_filepath, ParquetData& data_handler)
   if (!st.ok())
   {
     LOG_FAIL_FMT("Couldn't find generator file");
-    exit(2);
+    exit(1);
   }
   else
   {
     LOG_INFO_FMT("Found generator file");
   }
 
   // Read entire file as a single Arrow table
-  auto selected_columns = {0, 1};
-  std::shared_ptr<arrow::Table> table;
-  st = arrow_reader->ReadTable(selected_columns, &table);
-  if (!st.ok())
+  std::shared_ptr<arrow::Table> table = nullptr;
+  st = arrow_reader->ReadTable(&table);
+  if (!st.ok() || table == nullptr)
   {
-    LOG_FAIL_FMT("Couldn't open generator file");
-    exit(2);
+    LOG_FAIL_FMT("Couldn't open generator file: {}", st.ToString());
+    exit(1);
   }
   else
   {
     LOG_INFO_FMT("Opened generator file");
   }
 
-  std::shared_ptr<::arrow::ChunkedArray> column;
+  const auto& schema = table->schema();
+
+  std::vector<std::string> column_names = {"messageID", "request"};
 
-  ::arrow::Status column1Status = arrow_reader->ReadColumn(1, &column);

```

Comment:
We were previously parsing the file multiple times, first with `ReadTable` and then with `ReadColumn`. `arrow` is smart so that's still pretty fast, but to be safe I've moved to doing everything from the single table read. There's a bit more error checking/logging for failure cases, and some more resilience to schema changes/nearlycorrect files (looking up columns by name rather than assuming a known index).